.buildkite/pipeline.yml

```diff
@@ -23,42 +23,63 @@ steps:
             # Allow WP-CLI to be run as root, otherwise it throws an exception.
             # Reference: https://git.io/J9q2S
             - "WP_CLI_ALLOW_ROOT=true"
+      - peakon/git-shallow-clone#v0.0.1:
+          depth: 1
     command: |
         source /root/.bashrc
 
+        echo "--- :node: Setup Node environment"
         nvm install && nvm use
+
+        echo "--- :npm: Install Node dependencies"
         npm ci --unsafe-perm --prefer-offline --no-audit --no-progress
 
+        echo "--- :package: Run bundle prep work"
         npm run prebundle:js
+
+        echo "--- :android: Build Android bundle"
         npm run bundle:android
 
+        echo "--- :arrow_up: Upload Android bundle artifact"
         buildkite-agent artifact upload bundle/android/App.js
 
+        echo "--- :android: Build iOS bundle"
+        npm run bundle:ios
+
+        echo "--- :arrow_up: Upload iOS bundle artifact"
+        buildkite-agent artifact upload bundle/ios/App.js
+
   - label: "Build Android RN Aztec & Publish to S3"
     key: "publish-react-native-aztec-android"
     plugins:
       - *publish-android-artifacts-docker-container
+      - peakon/git-shallow-clone#v0.0.1:
+          depth: 1
     command: |
         .buildkite/publish-react-native-aztec-android-artifacts.sh
 
   - label: "Build Android RN Bridge & Publish to S3"
     depends_on:
-      - "bundle-android"
+      - "js-bundles"
       - "publish-react-native-aztec-android"
     plugins:
       - *publish-android-artifacts-docker-container
+      - peakon/git-shallow-clone#v0.0.1:
+          depth: 1
     command: |
         .buildkite/publish-react-native-bridge-android-artifacts.sh
 
   - label: Build iOS RN XCFramework
+    depends_on: js-bundles
     command: .buildkite/publish-react-native-ios-artifacts.sh
     artifact_paths:
+      - ios-xcframework/build/xcframeworks/*.tar.gz
       - ios-xcframework/build/xcframeworks/*.zip
     plugins:
-      - automattic/a8c-ci-toolkit#2.15.1
+      - automattic/a8c-ci-toolkit#2.16.0
       - peakon/git-shallow-clone#v0.0.1:
           depth: 1
     agents:
       queue: mac
     env:
-      IMAGE_ID: xcode-14.2
+      IMAGE_ID: xcode-14.3
```

Comment:
These updates were not required by the changes in the PR. I did them just to stay up to date.